core/src/main/java/com/adobe/aem/commons/assetshare/content/properties/impl/AssetRenditionImpl.java

```diff
@@ -0,0 +1,112 @@
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.adobe.aem.commons.assetshare.content.properties.impl;
+
+import com.adobe.aem.commons.assetshare.content.properties.AbstractComputedProperty;
+import com.adobe.aem.commons.assetshare.content.properties.ComputedProperty;
+import com.adobe.aem.commons.assetshare.content.renditions.impl.AssetRenditionServlet;
+import com.adobe.aem.commons.assetshare.util.MimeTypeHelper;
+import com.adobe.cq.commerce.common.ValueMapDecorator;
+import com.day.cq.dam.api.Asset;
+import com.day.cq.dam.api.Rendition;
+import com.day.cq.dam.commons.util.DamUtil;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.resource.ValueMap;
+import org.osgi.framework.Constants;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.metatype.annotations.AttributeDefinition;
+import org.osgi.service.metatype.annotations.Designate;
+import org.osgi.service.metatype.annotations.ObjectClassDefinition;
+
+import java.util.HashMap;
+
+import static com.adobe.aem.commons.assetshare.content.properties.ComputedProperty.DEFAULT_ASC_COMPUTED_PROPERTY_SERVICE_RANKING;
+
+@Component(
+        service = ComputedProperty.class,
+        property = {
+                Constants.SERVICE_RANKING + "=" + DEFAULT_ASC_COMPUTED_PROPERTY_SERVICE_RANKING
+        }
+)
+@Designate(ocd = AssetRenditionImpl.Cfg.class)
+public class AssetRenditionImpl extends AbstractComputedProperty<String> {
+
+    public static final String LABEL = "Asset Rendition";
+    public static final String NAME = "rendition";
+
+    private Cfg cfg;
+
+    @Override
+    public String getName() {
+        return NAME;
+    }
+
+    @Override
+    public String getLabel() {
+        return cfg.label();
+    }
+
+    @Override
+    public String[] getTypes() {
+        return cfg.types();
+    }
+
+    @Override
+    public String get(Asset asset, SlingHttpServletRequest request, ValueMap computedPropertyParameters) {
+        String url = asset.getPath() + "." + AssetRenditionServlet.SERVLET_EXTENSION + "/"
+                + computedPropertyParameters.get("name", String.class) + "/";
+
+        if (computedPropertyParameters.get("download", false)) {
+            url += "download/";
+        }
+
+        url += "asset.rendition";
+
+        return escapeString(url);
+    }
+
+    private String escapeString(String str) {
```

Comment:
Isn't there more to escaping a string besides these?

Reply:
Agh - let me sort out the Url escaping better/more centrally ... the escaping is a bit of a mess. 
This *works* but to you point, it's not obvious and looks incomplete due to other places escaping happens down the chain.